libs/entitlement/qa/common/src/main/java/org/elasticsearch/entitlement/qa/common/RestEntitlementsCheckAction.java

```diff
@@ -311,6 +322,70 @@ public DatagramSocketImpl createDatagramSocketImpl() {
         URLConnection.setContentHandlerFactory(__ -> { throw new IllegalStateException(); });
     }
 
+    private static void bindDatagramSocket() {
+        logger.info("Calling DatagramSocket#bind");
+        try (var socket = new DatagramSocket(null)) {
+            socket.bind(null);
+        } catch (SocketException e) {
+            throw new RuntimeException(e);
```

Comment:
Nit: I've been using `IllegalStateException` for these. `RuntimeException` tends to be a uniquely awkward exception to throw; for example, it's a superclass of `NullPointerException`. Really, `RuntimeException` should have been abstract, and I tend to treat it that way.

Reply:
I just used Intellij auto-generation. I can change the exception type, but it really does not matter - it should not throw, if it throws is a test error, an we just need it to fail to indicate the test is broken/wrong.

Reply:
I went with your suggestion in https://github.com/elastic/elasticsearch/pull/119867, so now we don't need to rethrow